crates/noirc_abi/src/lib.rs

```diff
@@ -66,6 +66,17 @@ impl AbiType {
         }
     }
 
+    /// Returns the number of field elements required to represent the type once encoded.
+    pub fn field_count(&self) -> usize {
+        match self {
+            AbiType::Field(_) | AbiType::Integer { .. } => 1,
+            AbiType::Array { visibility: _, length, typ } => typ.field_count() * (*length as usize),
```

Comment:
Apologies for not catching this earlier - if we only need `field_count` in one place expecting a `u32` we might as well return a `u32` directly. I'd rather convert to `u32` directly than having to convert to usize here then to u32 later.

Reply:
No worries. I wanted to keep `usize` for consistency with `AbiType.num_elements` but happy to change.